src/runtime/io.cpp

```diff
@@ -521,6 +524,194 @@ extern "C" LEAN_EXPORT obj_res lean_io_prim_handle_put_str(b_obj_arg h, b_obj_ar
+    using namespace std::chrono;
+#if defined(LEAN_WINDOWS)
+    UErrorCode status = U_ZERO_ERROR;
+    UCalendar *cal = ucal_open(NULL, -1, NULL, UCAL_GREGORIAN, &status);
+
+    if (U_FAILURE(status)) {
+        return lean_io_result_mk_error(lean_decode_io_error(EINVAL, mk_string("failed to open calendar")));
+    }
+
+    ucal_setMillis(cal, timestamp_secs * 1000, &status);
+    if (U_FAILURE(status)) {
+        ucal_close(cal);
+        return lean_io_result_mk_error(lean_decode_io_error(EINVAL, mk_string("failed to set calendar time")));
+    }
+
+    UChar tzId[256];
+    int32_t tzIdLength = ucal_getTimeZoneID(cal, tzId, sizeof(tzId)/sizeof(tzId[0]), &status);
+
+    if (U_FAILURE(status)) {
+        ucal_close(cal);
+        return lean_io_result_mk_error(lean_decode_io_error(EINVAL, mk_string("failed to get zone_offset")));
+    }
+
+    char dst_name[256];
+    u_strToUTF8(dst_name, sizeof(dst_name), NULL, tzId, tzIdLength, &status);
+    if (U_FAILURE(status)) {
+        ucal_close(cal);
+        return lean_io_result_mk_error(lean_decode_io_error(EINVAL, mk_string("failed to get zone_offset")));
+    }
+
+    UChar display_name[32];
+    int32_t display_name_len = ucal_getTimeZoneDisplayName(cal, UCAL_SHORT_STANDARD, "en_US", display_name, 32, &status);
+
+    if (U_FAILURE(status)) {
+        ucal_close(cal);
+        return lean_io_result_mk_error(lean_decode_io_error(EINVAL, mk_string("failed to read abbreaviation")));
+    }
+
+    char display_name_str[32];
+    u_strToUTF8(display_name_str, sizeof(display_name_str), NULL, display_name, display_name_len, &status);
+
+    if (U_FAILURE(status)) {
+        ucal_close(cal);
+        return lean_io_result_mk_error(lean_decode_io_error(EINVAL, mk_string("failed to get abbreviation to cstr")));
+    }
+
+    int32_t zone_offset = ucal_get(cal, UCAL_ZONE_OFFSET, &status);
+
+    if (U_FAILURE(status)) {
+        ucal_close(cal);
+        return lean_io_result_mk_error(lean_decode_io_error(EINVAL, mk_string("failed to get zone_offset")));
+    }
+
+    int32_t dst_offset = ucal_get(cal, UCAL_DST_OFFSET, &status);
+
+    zone_offset += dst_offset;
+    
+    if (U_FAILURE(status)) {
+        return lean_io_result_mk_error(lean_decode_io_error(EINVAL, mk_string("failed to get dst_offset")));
+    }
+
+    int offset_hour = zone_offset / 3600000;
+    int offset_seconds = zone_offset / 1000;
+    int is_dst = dst_offset != 0;
+#else
+    std::time_t input_time = static_cast<std::time_t>(timestamp_secs);
+    std::tm tm_info;
+    struct tm *tm_ptr = localtime_r(&input_time, &tm_info);
+
+    if (tm_ptr == NULL) {
+        return lean_io_result_mk_error(lean_decode_io_error(EINVAL, mk_string("")));
+    }
+    
+    int offset_hour = tm_info.tm_gmtoff / 3600;
+    int offset_seconds = tm_info.tm_gmtoff;
+    int is_dst = tm_info.tm_isdst;
+    char dst_name[] = "Unknown";
+    char display_name_str[32] = "Unknown";
+#endif
```

Comment:
Why implement this on non-Windows platforms? It looks like you only need this on Windows? But then why does the name not include `windows`?